controller/src/main/java/io/pravega/controller/store/stream/Stream.java

```diff
@@ -685,15 +685,19 @@
     CompletableFuture<Integer> getSegmentSealedEpoch(long segmentId);
 
     /**
-     * Method to compare streamcuts to check if streamcut1 is strictly ahead of streamcut2. 
-     * Strictly means if the two streamcuts are overlapping for any range, then this method will reply in negative. 
+     * Compares two Stream cuts and returns :
+     *  1 if streamcut1 is ahead of or identical to streamcut2. 
```

Comment:
> 1 if streamcut1 is ahead of or identical to streamcut2.
So `1` means S1 is strictly ahead of S2? or orverlaps with S2?... but overlap should be `0`...a bit confused about this...

Reply:
0: overlap means one streamcut has some segments where it is greater-than and on some it is less-than another streamcut. 
1: means streamcut is greater than or eq to another streamcut on all segments

Reply:
This is precisely why I think it may be better to use enums instead of 0,1 values, so we can better explain what each value implies.

Reply:
1) that enum will only be used for this method which is an internal method. 2) the javadoc of enum or the javadoc for the method signature will capture this exact same comment describing what each of the responses mean. so i do not see how enum may be better suited here. but i can make it an enum regardless.